android/app/src/main/java/org/jitsi/meet/MainActivity.java

```diff
@@ -1,46 +1,98 @@
+ */
+public class MainActivity extends AppCompatActivity {
+    private JKConferenceView conferenceView;
+    public static final int OVERLAY_PERMISSION_REQ_CODE = 4242;
+
     @Override
-    protected ReactActivityDelegate createReactActivityDelegate() {
-        return new ReactActivityDelegate(this, getMainComponentName()) {
-            /**
-             * {@inheritDoc}
-             *
-             * Overrides {@link ReactActivityDelegate#createRootView()} to
-             * customize the {@link ReactRootView} with a background color that
-             * is in accord with the JavaScript and iOS parts of the application
-             * and causes less perceived visual flicker than the default
-             * background color.
-             */
-            @Override
-            protected ReactRootView createRootView() {
-                ReactRootView rootView = super.createRootView();
-
-                rootView.setBackgroundColor(0xFF111111);
-                return rootView;
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == OVERLAY_PERMISSION_REQ_CODE) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                if (!Settings.canDrawOverlays(this)) {
+                    // Permission not granted...
+                    return;
+                }
             }
-        };
+
+            setContentView(conferenceView);
+        }
+    }
+
+
+    @Override
+    public void onBackPressed() {
+        if (!JKConferenceView.onBackPressed()) {
+            // Invoke the default handler if it wasn't handled by React.
+            super.onBackPressed();
+        }
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        conferenceView = new JKConferenceView(this);
+        conferenceView.loadURL(null);
+
+        /**
+         * In debug mode React needs permission to write over other apps in order to display the
+         * warning and error overlays.
+         */
+        if (BuildConfig.DEBUG && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
+            !Settings.canDrawOverlays(this)) {
+
+            Intent intent
+                = new Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION,
+                    Uri.parse("package:" + getPackageName()));
+            startActivityForResult(intent, OVERLAY_PERMISSION_REQ_CODE);
+            return;
+        }
+
+        setContentView(conferenceView);
     }
 
-    /**
-     * Returns the name of the main component registered from JavaScript.
-     * This is used to schedule rendering of the component.
-     */
     @Override
-    protected String getMainComponentName() {
-        return "App";
+    public void onNewIntent(Intent intent) {
+        JKConferenceView.onNewIntent(intent);
     }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        JKConferenceView.onHostDestroy(this);
```

Comment:
I do not understand why there are multiple static methods on `JKConferenceView` and I don't think that the javadocs on them really explain to me as a consumer of the API why I need to call them. My ideal Android View will not require such methods to be called by its consumer. If it's all about deep linking, cannot we extract linking in a separate API like React Native's LInking? I hardly think a non-trivial app will embed a Jitsi Meet view and agree for the app's linking to be entirely handled by the Jitsi Meet view.